lib/topic_view.rb

```diff
@@ -37,6 +37,27 @@ def self.whitelisted_post_custom_fields(user)
     wpcf.flatten.uniq
   end
 
+  def self.add_custom_filter(key, &blk)
+    @custom_filters ||= {}
+    @custom_filters[key] = blk
+  end
+
+  def self.remove_custom_filter(key)
+    @custom_filters.delete(key)
+    @custom_filters = nil if @custom_filters.length == 0
```

Comment:
Yeah that seems reasonable - there's probably no need for line 47 at all.

This stuff is mostly [taken from TopicQuery](https://github.com/discourse/discourse/blob/master/lib/topic_query.rb#L78) so I wonder if @SamSaffron had a reason for setting it to `nil`?